service/hproxy/hproxy.go

```diff
@@ -566,6 +601,26 @@ func (s *Server) handleProxyRequest(w http.ResponseWriter, r *http.Request) {
 //	}
 // }
 
+func (s *Server) filterRequest(r *http.Request) error {
+	// copy and reset body
+	const byteLimit = 5 * 1024 * 1024 // 5 MiB
+	lr := io.LimitReader(r.Body, byteLimit)
```

Comment:
If a request exceeds 5 MiB, the `json.NewDecoder` will likely fail to decode (e.g. unclosed `}`s). Do we want to add some kind of global request size limiter, and use that here? Then all requests exceeding that can be blocked 🤔